File: server/internal/models/game.go
package models

import "time"

type Game struct {
	dbModel
	Home           Team      `json:"home"`
	Away           Team      `json:"away"`
	HomeLockerRoom string    `json:"home_locker_room"`
	AwayLockerRoom string    `json:"away_locker_room"`
	Start          time.Time `json:"start"`
	End            time.Time `json:"end"`
	Location       Venue     `json:"location"`
	Staff          Staff     `json:"staff"`
	HomeScore      int       `json:"home_score"`
	AwayScore      int       `json:"away_score"`
}

File: frontend/models/Game.ts
import type { Team } from './Team'
import type { TimeString } from './types/timeString'
import type { Venue } from './Venue'

export interface Game {
  // dbModel: DbModel
  // season_id: SeasonId
  teams: Array<Team>
  home_locker_room: string
  away_locker_room: string
  start: TimeString
  end: TimeString
  venue: Venue
  // venue_id: VenueId

  // manager_on_call: User
  // // manager_on_call_id: UserId
  // score_keeper: User
  // primary_referee: User
  // secondary_referee: User

  // May be replaced with stats models
  // home_score: number
  // away_score: number
}
